src/telemetry/Manager.h

```diff
@@ -50,6 +57,73 @@ class Manager
 	 */
 	virtual void InitPostScript();
 
+	/**
+	 * Instances for use in __list_metrics, returned by by CollectMetrics().
+	 *
+	 * This is using the broker::telemetry low-level types rather
+	 * than IntCounter, DblCounter from zeek::telemetry due to the
+	 * latter expecting non-const pointers and I didn't want to
+	 * cast them away. The former provide a powerful enough API, too.
+	 */
```

Comment:
The `telemetry::IntGauge` (as an example) can be constructed with a non-const `broker::telemetry::int_gauge_hdl*` (the class offers Inc/Dec that mutates the underlying guage). The collector API only provides the const versions of these pointers (in the current form at least). I'm not sure what could be changed with the classes, but given that all we'd be using is `Value() on those, seemed okay to directly use the `broker::telemetry::value()` helper directly.

https://github.com/zeek/zeek/blob/318c50f0802958d8da81b114b1ddc220acbdb448/src/telemetry/Gauge.h#L78-L84

I think with your suggestion of  compressing the "CollectedMetric" it may look different enough to see where and whether it's a concern.

Reply:
I'm resolving this for now, but also pinging @Neverlord in the hope he can look over what's there currently.

Reply:
I couldn't find that comment in the current version. Is this still an open point?

Reply:
Maybe: For gathering the data, this still uses the new `collect` and the `broker::telemetry` free functions rather than `zeek::telemetry::DblGauge` or `zeek::telemetry::DblCounter` types. These are what you'd use primarily in C++ for actual metrics tracking/updating, so they may not be the best fit.

Haven't seen all the comments, but if that didn't stand out to you, too much it may be okay to keep it resolved.